src/org/ohdsi/usagi/ui/UsagiMain.java

```diff
@@ -180,4 +180,14 @@ private static Image loadIcon(String name, JFrame f) {
 		return null;
 	}
 
+	private Vector<String> loadVectorFromFile(String fileName) {
+		if (new File(fileName).exists()) {
+			Vector<String> vector = new Vector<>();
+			for (String line : new ReadTextFile(fileName))
+				vector.add(line);
+			return vector;
+		} else
+			return new Vector<>();
+	}
+
```

Comment:
If you don't absolutely need to use Vector (e.g. synchronised access to the array), use the following:
```suggestion
    private List<String> loadVectorFromFile(String fileName) {
        Path path = Paths.get(fileName);
        if (Files.exists(path)) {
            Files.newBufferedReader(path).lines()
                    .collect(Collections.toList());
        } else {
            return Collections.emptyList();
        }
    }
```
Otherwise, just wrap the return value I put above in a `new Vector<>(...)`.

Reply:
the original developer implemented this using `Vector`s. While it is probably not absolutely necessary, I don't want to risk changing it.